hawkbit-repository/hawkbit-repository-api/src/main/java/org/eclipse/hawkbit/repository/DistributionSetTypeManagement.java

```diff
@@ -106,74 +38,24 @@
-    Optional<DistributionSetType> findDistributionSetTypeByKey(@NotEmpty String key);
+    Optional<DistributionSetType> getByKey(@NotEmpty String key);
 
     /**
      * @param name
      *            as {@link DistributionSetType#getName()}
      * @return {@link DistributionSetType}
      */
     @PreAuthorize(SpringEvalExpressions.HAS_AUTH_READ_REPOSITORY)
-    Optional<DistributionSetType> findDistributionSetTypeByName(@NotEmpty String name);
-
-    /**
-     * @param pageable
-     *            parameter
-     * @return all {@link DistributionSetType}s in the repository.
-     */
-    @PreAuthorize(SpringEvalExpressions.HAS_AUTH_READ_REPOSITORY)
-    Page<DistributionSetType> findDistributionSetTypesAll(@NotNull Pageable pageable);
-
-    /**
-     * Generic predicate based query for {@link DistributionSetType}.
-     *
-     * @param rsqlParam
-     *            rsql query string
-     * @param pageable
-     *            parameter for paging
-     *
-     * @return the found {@link SoftwareModuleType}s
-     * 
-     * @throws RSQLParameterUnsupportedFieldException
-     *             if a field in the RSQL string is used but not provided by the
-     *             given {@code fieldNameProvider}
-     * @throws RSQLParameterSyntaxException
-     *             if the RSQL syntax is wrong
-     */
-    @PreAuthorize(SpringEvalExpressions.HAS_AUTH_READ_REPOSITORY)
-    Page<DistributionSetType> findDistributionSetTypesAll(@NotNull String rsqlParam, @NotNull Pageable pageable);
+    Optional<DistributionSetType> getByName(@NotEmpty String name);
 
     /**
      * @param id
      *            as {@link DistributionSetType#getId()}
      * @return {@link DistributionSetType}
      */
+    @Override
     @PreAuthorize(SpringEvalExpressions.HAS_AUTH_READ_REPOSITORY)
-    Optional<DistributionSetType> findDistributionSetTypeById(@NotNull Long id);
-
-    /**
-     * Updates existing {@link DistributionSetType}. Resets assigned
-     * {@link SoftwareModuleType}s as well and sets as provided.
-     *
-     * @param update
-     *            to update
-     * 
-     * @return updated entity
-     * 
-     * @throws EntityNotFoundException
-     *             in case the {@link DistributionSetType} does not exists and
-     *             cannot be updated
-     * 
-     * @throws EntityReadOnlyException
-     *             if the {@link DistributionSetType} is already in use by a
-     *             {@link DistributionSet} and user tries to change list of
-     *             {@link SoftwareModuleType}s
-     * @throws ConstraintViolationException
-     *             if fields are not filled as specified. Check
-     *             {@link DistributionSetTypeUpdate} for field constraints.
-     */
-    @PreAuthorize(SpringEvalExpressions.HAS_AUTH_UPDATE_REPOSITORY)
-    DistributionSetType updateDistributionSetType(@NotNull DistributionSetTypeUpdate update);
+    Optional<DistributionSetType> get(@NotNull Long id);
```

Comment:
Use same name for id parameter within the class -> for example distributionSetTypeId